ingress-controller/internal/manager/setup.go

```
@@ -533,3 +549,18 @@ func setupKonnectConfigSynchronizerWithMgr(
 	}
 	return s, nil
 }
+
+// buildLabelSelector creates a `label.Selector` from specified match tabels
+// to build the cache options used in the manager's `ByObject` to filter objects (Secrets, ConfigMaps) by labels.
+// TODO: move the code to build cache's ByObject options to common utils packages.
+func buildLabelSelector(matchLabels map[string]string) labels.Selector {
+	s := labels.NewSelector()
+	for k, v := range matchLabels {
+		req, err := labels.NewRequirement(k, selection.Equals, []string{v})
+		if err != nil {
+			continue
+		}
```

I'm not sure if silently ignoring this error is what we want. We're working here on user provided values and these are validated in https://github.com/kubernetes/apimachinery/blob/173776a0582da70432e19d16eca025c451600ae8/pkg/labels/selector.go#L209 via https://github.com/kubernetes/apimachinery/blob/v0.33.3/pkg/labels/selector.go#L923-L928.

WDYT about adding error propagation here?

---

Added error logs here.